packages/pluggableWidgets/rich-text-web/src/utils/ckeditorConfigs.ts

```diff
@@ -0,0 +1,82 @@
+import { CKEditorConfig } from "ckeditor4-react";
+import { PresetEnum, AdvancedConfigType, CtItemTypeEnum } from "../../typings/RichTextProps";
+import { createPreset, TOOLBAR_GROUP, ToolbarGroup, ToolbarItems } from "./ckeditorPresets";
+
+export type PluginName = "codesnippet";
+
+const PLUGIN_CONFIGS = {
+    codesnippet: {
+        extraPlugins: "codesnippet",
+        name: "CodeSnippet",
+        config: {
+            codeSnippet_theme: "idea"
+        }
+    }
+};
+
+export function getToolbarGroupByName(name: string): ToolbarGroup | undefined {
+    return TOOLBAR_GROUP.find((group: ToolbarGroup) => group.name === name);
+}
+
+export function defineEnterMode(type: string): number {
+    switch (type) {
+        case "paragraph":
+            return 1;
+        case "breakLines":
+            return 2;
+        case "blocks":
+            return 3;
+        default:
+            return 1;
+    }
+}
+
+export function getPreset(type: PresetEnum): CKEditorConfig {
+    switch (type) {
+        case "standard":
+            return createPreset("standard");
+        case "basic":
+            return createPreset("basic");
+        case "full":
+            return createPreset("full");
+        default:
+            return createPreset("basic");
+    }
+}
+
+export function addPlugin(name: PluginName, config: CKEditorConfig): CKEditorConfig {
+    const plugin = PLUGIN_CONFIGS[name];
+    if (plugin && config) {
+        if (config.extraPlugins && !config.extraPlugins.includes(plugin.extraPlugins)) {
+            config.extraPlugins += `,${plugin.extraPlugins}`;
+        } else {
+            config.extraPlugins = plugin.extraPlugins;
+        }
+    }
+    return config;
+}
+
+export function defineAdvancedGroups(items: AdvancedConfigType[]): ToolbarItems[] {
+    const toolbarObj: {
+        [key: string]: Array<CtItemTypeEnum | "-">;
+    } = {};
+    items.forEach(item => {
```

Comment:
Cant we map instead?

Reply:
Maps supposed to return something, here we just modifying object outside

Reply:
From functionality perspective, of course, it will work as expected, but from logical perspective it's better to `forEach` than `map` if you don't expect `return`